### PR TITLE
PIMS-1660 (SPP Property Filter) / PIMS-2650 (CSV/Excel Export)

### PR DESCRIPTION
**Pull request by jdtoombs**
Will probably have to refactor after UI, but want to make sure I am approaching this properly

### CODE REVIEW

File: backend/dal/Helpers/Extensions/PropertyExtensions.cs [hunk 1]
```diff
@@ -70,6 +70,8 @@ public static IQueryable<Entity.Views.Property> GenerateQuery(this PimsContext c
                 query = query.Where(p => EF.Functions.Like(p.ProjectNumber, $"{filter.ProjectNumber}%"));
             if (filter.IgnorePropertiesInProjects == true)
                 query = query.Where(p => p.ProjectNumber == null);
+            if(filter.ProjectPropertiesOnly == true)
+                query = query.Where(p => p.ProjectNumber.Contains("SPP"));
```
Comment (jdtoombs) [inline backend/dal/Helpers/Extensions/PropertyExtensions.cs:74]: this is what I am mainly wondering about, is this sufficient?

File: frontend/src/features/projects/list/ProjectListView.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import './ProjectListView.scss';
```
Comment (jdtoombs) [inline frontend/src/features/projects/list/ProjectListView.tsx:1]: I am currently adding more filters in for the reports to match the acceptance criteria - leaving PR as draft until then

File: backend/entities/Models/PropertyFilter.cs [hunk 3]
```diff
@@ -46,6 +46,12 @@ public abstract class PropertyFilter : PageFilter
         /// <value></value>
         public bool? IgnorePropertiesInProjects { get; set; }
 
+        /// <summary>
+        /// get/set - Flag indicating to show only properties that belong to a project.
+        /// </summary>
+        /// <value></value>
+        public bool? ProjectPropertiesOnly { get; set; }
```
Comment (Fosol) [inline backend/entities/Models/PropertyFilter.cs:53]: I don't think this is required.

File: backend/dal/Helpers/Extensions/PropertyExtensions.cs (hunk 1 above)
Comment (Fosol) [reply backend/dal/Helpers/Extensions/PropertyExtensions.cs:74]: I think the following is more correct.  Generally returning anything with project number and has been approved.
`!String.IsNullOrWhitespace(Property.ProjectNumber) && !property.Projects.Any(p => p.IsProjectInDraft() || !p.IsProjectClosed())`

File: backend/api/Areas/Property/Models/Search/PropertyFilterModel.cs [hunk 4]
```diff
@@ -66,6 +66,13 @@ public class PropertyFilterModel : PageFilter
         /// <value></value>
         public bool? IgnorePropertiesInProjects { get; set; }
 
+        /// <summary>
+        /// get/set - Flag indicating to show only properties that belong to a project.
+        /// </summary>
+        /// <value></value>
+
+        public bool? ProjectPropertiesOnly { get; set; }
```
Comment (Fosol) [inline backend/api/Areas/Property/Models/Search/PropertyFilterModel.cs:74]: Probably should rename to `InSurplusPropertyProgram`

File: frontend/src/features/projects/list/ProjectListView.tsx [hunk 5]
```diff
@@ -32,6 +35,9 @@ interface IProjectFilterState {
   assessWorkflow?: boolean;
 }
 
+const getPropertyReportUrl = (filter: IProjectFilter) =>
```
Comment (Fosol) [inline frontend/src/features/projects/list/ProjectListView.tsx:38]: This is a project report, not a property report.

Comment (Fosol) [reply frontend/src/features/projects/list/ProjectListView.tsx:38]: Additionally you'll need to create a way to download two types of Excel documents.  The first is the general project report, the second is the Surplus Property List report.

1) Generic Project List - GET `/api/reports/projects` or POST `/api/reports/projects/filter`
2) Maggie's Financial Report - GET `/api/reports/projects/surplus/properties/list` or POST `/api/reports/projects/surplus/properties/list/filter`

The first one is essentially identical to what we did for the Properties List View, whereas the second is to all Maggie to continue using her Excel spreadsheet for reporting.

File: frontend/src/features/projects/list/ProjectListView.tsx [hunk 6]
```diff
@@ -146,11 +162,29 @@ const ProjectListView: React.FC<IProps> = ({ filterable, title, mode }) => {
     },
     [setData, setPageCount, mode],
   );
-
+  const dispatch = useDispatch();
   // Listen for changes in pagination and use the state to fetch our new data
   useEffect(() => {
     fetchData({ pageIndex, pageSize, filter, agencyIds });
-  }, [fetchData, pageIndex, pageSize, filter, agencyIds]);
+    dispatch(fetchProjectStatuses());
```
Comment (jdtoombs) [inline frontend/src/features/projects/list/ProjectListView.tsx:169]: cleaning this up right now

File: frontend/src/features/projects/list/ProjectListView.tsx [hunk 7]
```diff
@@ -224,6 +258,18 @@ const ProjectListView: React.FC<IProps> = ({ filterable, title, mode }) => {
                 <Field name="createdByMe" type="checkbox" />
               </Form.Group>
             </Col>
+            <Col className="bar-item">
+              {/* Will need UI for this before merging */}
+              <Select field="statusId" options={statuses} multiple={true} label="Project Status:" />
```
Comment (jdtoombs) [inline frontend/src/features/projects/list/ProjectListView.tsx:263]: Probably have to make a separate component based of the UI

**Pull request closed without merging**